rmw_implementation/src/functions.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FUNCTIONS_HPP_
```

Comment:
I think this will run afoul of the linters, but maybe they don't care about private headers?  (I don't care either way)

Reply:
It was actually `cpplint` that forced me to use this guard.

Reply:
> It was actually `cpplint` that forced me to use this guard.

Haha, OK.